integration/common/openlineage/common/provider/dbt.py

```diff
@@ -0,0 +1,202 @@
+            ] + [
+                manifest['sources'][source]
+                for source
+                in manifest['parent_map'][run['unique_id']]
+                if source.startswith('source.')
+            ]
+
+            runs.append(DbtRun(
+                started_at,
+                completed_at,
+                run['status'],
+                inputs,
+                nodes[run['unique_id']],
+                run['unique_id'],
+                self.namespace
+            ))
+        return runs
+
+    def to_openlineage_events(self, run: DbtRun) -> Tuple[RunEvent, RunEvent]:
+        try:
+            return RunEvent(
+                eventType=RunState.START,
+                eventTime=run.started_at,
+                run=Run(
+                    runId=run.run_id
+                ),
+                job=Job(
+                    namespace=self.namespace,
+                    name=run.job_name
+                ),
+                producer=PRODUCER,
+                inputs=[self.node_to_dataset(node) for node in run.inputs],
+                outputs=[self.node_to_dataset(run.output)]
+            ), RunEvent(
+                eventType=RunState.COMPLETE,
+                eventTime=run.completed_at,
+                run=Run(
+                    runId=run.run_id
+                ),
+                job=Job(
+                    namespace=self.namespace,
+                    name=run.job_name,
+                    facets={
+                        'sql': SqlJobFacet(run.output['compiled_sql'])
+                    }
+                ),
+                producer=PRODUCER,
+                inputs=[self.node_to_dataset(node, facets=True) for node in run.inputs],
+                outputs=[self.node_to_dataset(run.output, facets=True)]
+            )
+        except Exception as e:
+            raise ValueError(e)
+
+    def node_to_dataset(self, node: Dict, facets: bool = False) -> Dataset:
+        if facets:
+            facets = {
+                'dataSource': DataSourceDatasetFacet(
+                    name=self.namespace,
+                    uri=self.namespace
+                ),
+                'schema': SchemaDatasetFacet(
+                    fields=[SchemaField(
+                        name=field['name'], type=field['data_type']
+                    ) for field in node['columns'].values()]
+                )
+            }
+        else:
+            facets = {}
+        return Dataset(
+            namespace=self.namespace,
+            name=f"{node['database']}.{node['schema']}.{node['name']}",
+            facets=facets
+        )
+
+    def set_namespace(self, profile: Dict):
+        if profile['type'] == 'snowflake':
+            self.namespace = f"snowflake://{profile['account']}"
+        elif profile['type'] == 'bigquery':
+            self.namespace = "bigquery"
+        else:
+            raise NotImplementedError(
+                "Only 'snowflake' and 'bigquery' adapters are supported right now."
```

Comment:
Can you also log the unsupported type in the error message?

Reply:
Done.